libusb/os/linux_usbfs.h

```diff
@@ -164,10 +164,12 @@ int linux_udev_start_event_monitor(void);
 int linux_udev_stop_event_monitor(void);
 int linux_udev_scan_devices(struct libusb_context *ctx);
 void linux_udev_hotplug_poll(void);
-#else
+#elif !defined(__ANDROID__)
 int linux_netlink_start_event_monitor(void);
 int linux_netlink_stop_event_monitor(void);
 void linux_netlink_hotplug_poll(void);
+#elif defined(__ANDROID__)
```

Comment:
These function prototypes inside the header file are wrapped in a conditional. Should the functions itself in the 'libusb/os/linux_netlink.c' file be wrapped out to the same conditional?

I have these errors if build with NDK 25.2.9519653.
```shell
libusb/os/linux_netlink.c:94:5: error: no previous prototype for function 'linux_netlink_start_event_monitor'
int linux_netlink_start_event_monitor(void)

libusb/os/linux_netlink.c:155:5 error: no previous prototype for function 'linux_netlink_stop_event_monitor'
int linux_netlink_stop_event_monitor(void)

libusb/os/linux_netlink.c:397:6 error: no previous prototype for function 'linux_netlink_hotplug_poll'
void linux_netlink_hotplug_poll(void)
```

Reply:
I might consider removing the netlink file from the build entirely on line 37 of android/jni/libusb.mk, given android is an uncommon build target. I do not know netlink's potential role on rooted devices but it looks disabled as of now. Adding the proposed guards or changing build flags such that the condition is not an error would both also work.